website/content/docs/v0.12/Guides/air-gapped.md

```diff
@@ -0,0 +1,137 @@
+```
+
+This registry will be accepting connections on port 6000 on the host IPs.
+The registry is empty by default, so we have fill it with the images required by Talos.
+
+First, we pull all the images to our local Docker daemon:
+
+```bash
+$ for image in `talosctl images`; do docker pull $image; done
+v0.12.0-amd64: Pulling from coreos/flannel
+Digest: sha256:6d451d92c921f14bfb38196aacb6e506d4593c5b3c9d40a8b8a2506010dc3e10
+...
+```
+
+All images are now stored in the Docker daemon store:
+
+```bash
+$ docker images
+ghcr.io/talos-systems/install-cni    v0.3.0-12-g90722c3      980d36ee2ee1        5 days ago          79.7MB
+k8s.gcr.io/kube-proxy-amd64          v1.20.0                 33c60812eab8        2 weeks ago         118MB
+...
+```
+
+Now we need to re-tag them so that we can push them to our local registry.
+We are going to replace the first component of the image name (before the first slash) with our registry endpoint `127.0.0.1:6000`:
+
+```bash
+$ for image in `talosctl images`; do \
+    docker tag $image `echo $image | sed -E 's#^[^/]+/#127.0.0.1:6000/#'` \
+  done
+```
+
+As the next step, we push images to the internal registry:
+
+```bash
+$ for image in `talosctl images`; do \
+    docker push `echo $image | sed -E 's#^[^/]+/#127.0.0.1:6000/#'` \
+  done
+```
+
+We can now verify that the images are pushed to the registry:
+
+```bash
+$ curl  http://127.0.0.1:6000/v2/_catalog
+{"repositories":["autonomy/kubelet","coredns","coreos/flannel","etcd-development/etcd","kube-apiserver-amd64","kube-controller-manager-amd64","kube-proxy-amd64","kube-scheduler-amd64","talos-systems/install-cni","talos-systems/installer"]}
+```
+
+> Note: images in the registry don't have the registry endpoint prefix anymore.
+
+## Launching Talos in an Air-gapped Environment
+
+For Talos to use the internal registry, we use the registry mirror feature to redirect all the image pull requests to the internal registry.
+This means that the registry endpoint (as the first component of the image reference) gets ignored, and all pull requests are sent directly to the specified endpoint.
+
+We are going to use a QEMU-based Talos cluster for this guide, but the same approach works with Docker-based clusters as well.
+As QEMU-based clusters go through the Talos install process, they can be used better to model a real air-gapped environment.
+
+The `talosctl cluster create` command provides conveniences for common configuration options.
+The only required flag for this guide is `--registry-mirror '*'=http://10.5.0.1:6000` which redirects every pull request to the internal registry.
+The endpoint being used is `10.5.0.1`, as this is the default bridge interface address which will be routable from the QEMU VMs (`127.0.0.1` IP will be pointing to the VM itself).
+
+```bash
+$ sudo -E talosctl cluster create --provisioner=qemu --registry-mirror '*'=http://10.5.0.1:6000 --install-image=ghcr.io/talos-systems/installer:v0.11.0
```

Comment:
Should be updated to 0.12 I suppose.

Reply:
That PR mostly creates the folder. Updating for 0.12 is the next PR.

Reply:
gotcha